src/libraries/System.Text.RegularExpressions/tests/UnitTests/SymbolicRegexTests.cs

```diff
@@ -253,5 +253,19 @@ public void SafeThresholdConfigTest(object? newThresholdData, int expectedThresh
             AppContext.SetData(SymbolicRegexThresholds.SymbolicRegexSafeSizeThreshold_ConfigKeyName, null);
             Assert.Equal(expectedThreshold, k);
         }
+
+        [Fact]
+        public static void OptimizedReversalTests()
+        {
+            var charSetSolver = new CharSetSolver();
+            var bddBuilder = new SymbolicRegexBuilder<BDD>(charSetSolver, charSetSolver);
+            var converter = new RegexNodeConverter(bddBuilder, null);
+            const RegexOptions options = RegexOptions.NonBacktracking | RegexOptions.ExplicitCapture;
+            RegexNode tree = RegexParser.Parse("abc.*def", options, CultureInfo.CurrentCulture).Root;
+            SymbolicRegexNode<BDD> rootNode = converter.ConvertToSymbolicRegexNode(tree);
+            // todo: import the matcher here or use something else?
+            // var matcher = SymbolicRegexMatcher.Create(bddBuilder, rootNode, 0, null, TimeSpan.MaxValue);
```

Comment:
The amount of references required to test `SymbolicRegexMatcher` seemed too large so i left it there at the moment because i was unsure if it was the correct place. I was planning to write some simple test cases for `CreateOptimizedReversal`.

Can you give me an example of a test using `SymbolicRegexMatcher`?

Reply:
Do we need to test it directly, or can we test it indirectly just by adding whatever patterns and inputs would be necessary to exercise it in the FunctionalTests project?

Reply:
This is effectively covered by almost every single that has fixed length parts. I suppose it is fine to cover it indirectly as well